plugins/node/instrumentation-undici/src/undici.ts

```diff
@@ -139,12 +140,35 @@ export class UndiciInstrumentation extends InstrumentationBase<UndiciInstrumenta
     diagnosticChannel: string,
     onMessage: ListenerRecord['onMessage']
   ) {
-    const channel = diagch.channel(diagnosticChannel);
-    channel.subscribe(onMessage);
+    // NOTE: later versions of `@types/node` include proper type definitions
+    // for diagnostics channel. However updating it breaks compilation :(
+    // Hence we have a helper interface to deal with it
+    const diagchApi = diagch as unknown as DiagnosticChannelApi;
+
+    // NOTE: diagnostic channel had an issue tht was solved in v18.19.0 so
+    // instrumentation is going to use the new API when the process version is
+    // greater or equal
+    // ref: https://github.com/nodejs/node/pull/47520
+    const [major, minor] = process.version
+      .replace('v', '')
+      .split('.')
+      .map(n => Number(n));
+    const isCompatible = major > 18 || (major === 18 && minor >= 19);
+
+    let unsubscribe: () => void;
+    if (typeof diagchApi.subscribe === 'function' && isCompatible) {
```

Comment:
```suggestion
    const useNewSubscribe = major > 18 || (major === 18 && minor >= 19);

    let unsubscribe: () => void;
    if (useNewSubscribe) {
```

nit: I think you could drop the `typeof diagchApi.subscribe === 'function'` check, now that it is being gated on being >=18.19.